src/Features/Core/Portable/QualifyMemberAccess/AbstractQualifyMemberAccessDiagnosticAnalyzer.cs

```diff
@@ -48,7 +45,7 @@ public override bool OpenFileOnly(Workspace workspace)
         protected abstract bool IsAlreadyQualifiedMemberAccess(SyntaxNode node);
 
         protected override void InitializeWorker(AnalysisContext context)
-            => context.RegisterOperationAction(AnalyzeOperation, OperationKind.FieldReference, OperationKind.PropertyReference, OperationKind.MethodReference);
+            => context.RegisterOperationAction(AnalyzeOperation, OperationKind.FieldReference, OperationKind.PropertyReference, OperationKind.MethodReference, OperationKind.Invocation);
```

Comment:
1. wrap plz.
2. this seems odd.  why do you need to explicitly handle invocation operations?  an invocation will be of the form "expr(...)".  why is it not sufficient to just process hte "expr" part and handle it if it is some sort of member reference?

Reply:
Yes, this is unfortunately a bug that creeped into IOperation where [IInvocationOperation](http://source.roslyn.io/#Microsoft.CodeAnalysis/Operations/IInvocationOperation.cs,28) directly exposes the instance instead of the an `IMethodReference` for the method being invoked. @AlekseyTs @333fred as FYI. I am not sure if we can fix this without a breaking change.

Reply:
Hrm.. in the case of things just being buggy, i think it might be good to just fix, document as a change in behavior and bite the bullet early before IOperation spreads out further and further.

Reply:
Also tagging @jinujoseph. I will file a bug to track this issue.

This also needs an API change as we probably have to replace [IInvocationOperation.Instance](http://source.roslyn.io/#Microsoft.CodeAnalysis/Operations/IInvocationOperation.cs,28) with `IMethodReferenceOperation MemberReference`. Note this will break the existing analyzers based on 2.6 APIs that either use `IInvocationOperation.Instance` and/or rely on the tree shape for invocation operations.

Reply:
https://github.com/dotnet/roslyn/issues/26206

Reply:
Best would probably be to go through teh compat council and get their decision.  I know we've made 'breaking changes' to semantic model APis in hte past when they returned incorrect data.  i'm hoping we can do the same here.  @mavasani do we have a bug on this?  if not, can you open one?

Reply:
I don't think there is a bug in exposing an instance from IInvocationOperation. See https://github.com/dotnet/roslyn/issues/23084 for an alternative solution.

Reply:
@CyrusNajmabadi, @mavasani 
>This was addressed awhile back, but I forgot to close this issue. As part of this, MethodReferenceExpression was renamed to MethodBindingExpression, and occurs in delegate creation. Normal method invocation does not produce an expression for the method reference, but one can get it from the invocation expression.

https://github.com/dotnet/roslyn/issues/7584#issuecomment-179461699

Reply:
@zaytsev-victor Note: the IOperation from back in 2015/2016 is basically unrelated to the current IOperation :)  The one back then was a very rough prototype, and hte current one was basically redone and totally fleshed out.

Reply:
@CyrusNajmabadi should I close it and wait for #26206?

Reply:
No no.  Please leave open.  Let's see what the decision is.  If we decide to not really change the shape of IOperation, we'll need this PR where you ergister for OperationKind.Invocation.  if the shape is changed, then we'll still want this PR so we can unskip the tests.  In either event, let's just put this in a holding pattern for now until other decisions are made.

Reply:
I agree with @CyrusNajmabadi - this is a correct fix for the current IOperation API/tree shape and no fix to #26026 should (ideally) break this implementation.

Reply:
@mavasani I assume you made a mistake with the issue number?

Reply:
@Neme12 Yes, fixed it now, thanks.